File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.swift
import Foundation

func solution(_ numbers:[Int], _ target:Int) -> Int {
    var answer = 0
    
    func dfs(_ i:Int, _ total:Int, _ answer:inout Int) {
        if i == numbers.count {
            if total == target {
                answer += 1
            }
            return
        }
        dfs(i+1,total+numbers[i],&answer)
        dfs(i+1,total-numbers[i],&answer)
        return
    }
    
    dfs(0,0,&answer)
    return answer
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    global answer
    answer = 0
    def dfs(i, total):        
        global answer
        if i == len(numbers):
            if total == target:
                answer += 1
            return
        dfs(i+1, total+numbers[i])
        dfs(i+1, total-numbers[i])
        return
    dfs(0,0)
    return answer